client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -11,36 +11,83 @@ import { connect } from 'react-redux';
 /**
  * Internal dependencies
  */
+import { activateModule, deactivateModule } from 'state/jetpack/modules/actions';
 import Card from 'components/card';
-import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
 import FormFieldset from 'components/forms/form-fieldset';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug, isJetpackMinimumVersion } from 'state/sites/selectors';
+import isActivatingJetpackModule from 'state/selectors/is-activating-jetpack-module';
+import isDeactivatingJetpackModule from 'state/selectors/is-deactivating-jetpack-module';
+import isJetpackModuleActive from 'state/selectors/is-jetpack-module-active';
 import isJetpackModuleUnavailableInDevelopmentMode from 'state/selectors/is-jetpack-module-unavailable-in-development-mode';
 import isJetpackSiteInDevelopmentMode from 'state/selectors/is-jetpack-site-in-development-mode';
-import { getSelectedSiteId } from 'state/ui/selectors';
-import { getSiteSlug } from 'state/sites/selectors';
+import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
 import SupportInfo from 'components/support-info';
 
 class SpeedUpSiteSettings extends Component {
+	static defaultProps = {
+		togglingSiteAccelerator: false,
+	};
+
 	static propTypes = {
-		fields: PropTypes.object,
+		activateModule: PropTypes.func,
+		deactivateModule: PropTypes.func,
 		isRequestingSettings: PropTypes.bool,
 		isSavingSettings: PropTypes.bool,
 		jetpackVersionSupportsLazyImages: PropTypes.bool,
+		submitForm: PropTypes.func.isRequired,
+		updateFields: PropTypes.func.isRequired,
 
 		// Connected props
+		assetCdnModuleActive: PropTypes.bool,
+		photonModuleActive: PropTypes.bool,
 		photonModuleUnavailable: PropTypes.bool,
 		selectedSiteId: PropTypes.number,
+		siteAcceleratorStatus: PropTypes.bool,
+		siteAcceleratorSupported: PropTypes.bool,
 		siteSlug: PropTypes.string,
+		togglingSiteAccelerator: PropTypes.bool,
+	};
+
+	handleCdnChange = () => {
+		const { siteAcceleratorStatus, submitForm, updateFields } = this.props;
+
+		// If one of them is on, we turn everything off.
+		if ( siteAcceleratorStatus ) {
```

Comment:
It seems we can shorten:

```
if ( siteAcceleratorStatus ) {
	updateFields(
		{
			photon: false,
			'photon-cdn': false,
		},
		() => {
			submitForm();
		}
	);
} else {
	updateFields(
		{
			photon: true,
			'photon-cdn': true,
		},
		() => {
			submitForm();
		}
	);
}
```

to:

```
updateFields(
	{
		photon: ! siteAcceleratorStatus,
		'photon-cdn': ! siteAcceleratorStatus,
	},
	() => {
		submitForm();
	}
);
```

Reply:
This does look way better indeed. Fixed in 00dece7b0e5e3e56410008a18f425d1b48f9df96